cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,182 @@
+        fprintf(out, " %-*s\n", static_cast<int>(CommandExecutor::kInitMaxColumnWidth), "table");
+      }
+    } else {
+      fprintf(out, " %-*s|", static_cast<int>(max_column_width), relation->getName().c_str());
+      fprintf(out, " %-*s\n", static_cast<int>(CommandExecutor::kInitMaxColumnWidth), "table");
+    }
+    fputc('\n', out);
+  }
+}
+
+void CommandExecutor::executeDescribeTable(
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase &catalog_database, FILE *out) {
+  const ParseString &table_name = arguments->front();
+  const std::string &table_name_val = table_name.value();
+  const CatalogRelation *relation =
+      catalog_database.getRelationByName(table_name_val);
+  if (relation == nullptr) {
+     THROW_SQL_ERROR_AT(&(arguments->front())) << " Unrecognized relation "  <<table_name_val;
+  }
+  vector<std::size_t> column_widths;
+  std::size_t max_attr_column_width = CommandExecutor::kInitMaxColumnWidth;
+  std::size_t max_type_column_width = CommandExecutor::kInitMaxColumnWidth;
+
+  for (const CatalogAttribute &attr : *relation) {
+    // Printed column needs to be wide enough to print:
+    //   1. The attribute name (in the printed "header").
+    //   2. Any value of the attribute's Type.
+    std::size_t attr_column_width = attr.getDisplayName().length();
+    std::size_t type_column_width = attr.getType().getName().length();
+    max_attr_column_width = std::max(max_attr_column_width, attr_column_width);
+    max_type_column_width = std::max(max_type_column_width, type_column_width);
+  }
+  column_widths.push_back(max_attr_column_width);
+  column_widths.push_back(max_type_column_width);
+
+  fprintf(out, "%*s \"%s\"\n", static_cast<int>(CommandExecutor::kInitMaxColumnWidth), "Table", table_name_val.c_str());
+  fprintf(out, "%-*s|", static_cast<int>(max_attr_column_width)+1, " Column");
+  fprintf(out, "%-*s\n", static_cast<int>(max_type_column_width)+1, " Type");
+  PrintToScreen::printHBar(column_widths, out);
+  for (const CatalogAttribute &attr : *relation) {
+    fprintf(out, " %-*s|", static_cast<int>(max_attr_column_width),
+            attr.getDisplayName().c_str());
+    fprintf(out, " %-*s\n", static_cast<int>(max_type_column_width),
+            attr.getType().getName().c_str());
+  }
+  // TODO(rogers): Add handlers for partitioning information.
+  if (relation->hasIndexScheme()) {
+    fprintf(out, "%*s\n", static_cast<int>(CommandExecutor::kInitMaxColumnWidth)+2, " Indexes");
+    const quickstep::IndexScheme &index_scheme = relation->getIndexScheme();
+    for (auto index_it = index_scheme.begin(); index_it != index_scheme.end();
+         ++index_it) {
+      fprintf(out, "  \"%-*s\" %s", static_cast<int>(index_it->first.length()),
+              index_it->first.c_str(),
+              index_it->second.IndexSubBlockType_Name(
+                                  index_it->second.sub_block_type())
```

Comment:
Please fix the indentation like this:

```
  index_it->second.IndexSubBlockType_Name(
      index_it->second.sub_block_type());
```

Note that we add four whitespace indentations when wrapping an expression, while two for a complete statement ended with `;`.